x-attoparsec/src/X/Data/Attoparsec/Text.hs

```diff
@@ -31,8 +31,7 @@ positiveIntParser :: Parser Int
 positiveIntParser =
   (P.readMaybe <$> many1 digit) >>= P.maybe (fail "not a positive int") pure
 
--- | create a Parser a from a text Parser
-eitherText :: (T.Text -> Either T.Text a) -> Parser a
-eitherText p = do
-  t <- takeText
-  either fail pure (first T.unpack (p t))
+-- | create a Parser a from a text Parser and an input text
+--   usage: takeWhile (/= '-') >>= eitherText parseTimestamp
+eitherText :: (T.Text -> Either T.Text a) -> T.Text -> Parser a
+eitherText p = either (fail . T.unpack) pure . p
```

Comment:
Give that you're just passing in the function + text and running that function on the text I would make this the more general

`Either T.Text a -> Parser a`